pkg/sql/upsert.go

```diff
@@ -135,22 +136,34 @@ func (p *planner) makeUpsertHelper(
 	}
 	helper.evalExprs = evalExprs
 
-	helper.allExprsIdentity = true
+	helper.allUpdateExprsIdentity = true
 	for _, expr := range evalExprs {
 		ivar, ok := expr.(*parser.IndexedVar)
 		if !ok {
-			helper.allExprsIdentity = false
+			helper.allUpdateExprsIdentity = false
```

Comment:
I think there is a condition missing here. What about `SET x = excluded.y, y = excluded.x`. Both are `IndexedVars` and in the right range, yet they are not preserving the value. What do you think?